src/lib/Wst/Offchain/Env.hs

```diff
@@ -355,14 +355,15 @@ withTransfer dir action = do
   asks (addTransferEnv dir)
     >>= runReaderT action
 
-withTransferFor :: MonadReader (CombinedEnv o d t r era) m => C.Hash C.PaymentKey -> ReaderT (CombinedEnv o d Identity r era) m a -> m a
-withTransferFor = withTransfer . mkTransferLogicEnv
+withTransferFor :: MonadReader (CombinedEnv o d t r era) m => C.PaymentCredential -> C.Hash C.PaymentKey -> ReaderT (CombinedEnv o d Identity r era) m a -> m a
```

Comment:
If you change the `d` parameter to `Identity` you could get the payment credential from `DirectoryEnv`.